nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/test/java/org/apache/nifi/processors/standard/TestUnpackContent.java

```diff
@@ -221,6 +226,32 @@ public void testInvalidZip() throws IOException {
         }
     }
 
+    @Test
+    public void testZipEncryptionZipStandard() throws IOException {
+        runZipEncryptionMethod(EncryptionMethod.ZIP_STANDARD);
+    }
+
+    @Test
+    public void testZipEncryptionAes() throws IOException {
+        runZipEncryptionMethod(EncryptionMethod.AES);
+    }
+
+    @Test
+    public void testZipEncryptionNoPasswordConfigured() throws IOException {
+        final TestRunner runner = TestRunners.newTestRunner(new UnpackContent());
+        runner.setProperty(UnpackContent.PACKAGING_FORMAT, UnpackContent.PackageFormat.ZIP_FORMAT.toString());
+
+        final String password = String.class.getSimpleName();
+        final char[] streamPassword = password.toCharArray();
+        final String contents = TestRunner.class.getCanonicalName();
+
+        final byte[] zipEncrypted = createZipEncrypted(EncryptionMethod.AES, streamPassword, contents);
```

Comment:
Would be good to have a test method where you use something like DES/Triple DES to prove the error handling on unsupported algorithms.

Reply:
Thanks for the additional feedback.  The EncryptionMethod enum lists only supported algorithms, so testing a different type of encrypted zip file would require creating an example from another program.  Creating an encrypted Zip file with a trial version of PKWARE SecureZIP using the 3DES option results in the ZipInputStream throwing a ZipException with the following message:

"Reached end of entry, but crc verification failed for testing"

Unfortunately this doesn't provide any particular indication that the Zip entry was encrypted using some other unsupported algorithm.  The end result is that the file is routed to failure, just as with any other ZipException.

Implementing this unit test would require checking in an encrypted binary created with PKWARE SecureZIP.  Since the exception handling follows the same path as a missing or incorrect password, do you still want this test case included?

Reply:
Further review of Zip4j indicates that the only notable improvement to processing unsupported algorithms would be better exception messages from that library.  Since there a number of possible unsupported algorithms that would result in some kind of ZipException or other exception from Zip4j, there doesn't seem to be much value to exercising the same path through the code with unsupported binary inputs.

Reply:
I think we can skip it

Reply:
Thanks for the confirmation.